File: hello-dyn-csharp/CalculatorImplementations.cs
using System;

namespace DynamicCalculator
{
    // 基础计算器实现
    public class BasicCalculator : ICalculator
    {
        public double AddFn(double a, double b)
        {
            return a + b;
        }

        public double SubFn(double a, double b)
        {
            return a - b;
        }

        public double MulFn(double a, double b)
        {
            return a * b;
        }

        public double DivFn(double a, double b)
        {
            if (b == 0)
            {
                throw new CalculatorException("除数不能为零");
            }
            return a / b;
        }
    }

    // 高级计算器实现
    public class AdvancedCalculator : ICalculator
    {
        public double AddFn(double a, double b)
        {
            Console.WriteLine($"[ADVANCED] 加法: {a:F2} + {b:F2}");
            return a + b;
        }

        public double SubFn(double a, double b)
        {
            Console.WriteLine($"[ADVANCED] 减法: {a:F2} - {b:F2}");
            return a - b;
        }

        public double MulFn(double a, double b)
        {
            Console.WriteLine($"[ADVANCED] 乘法: {a:F2} * {b:F2}");
            return a * b;
        }

        public double DivFn(double a, double b)
        {
            if (b == 0)
            {
                throw new CalculatorException("除数不能为零");
            }
            Console.WriteLine($"[ADVANCED] 除法: {a:F2} / {b:F2}");
            return a / b;
        }

        // 额外功能
        public double PowFn(double baseValue, double exponent)
        {
            return Math.Pow(baseValue, exponent);
        }
    }
}

File: hello-dyn-typescript/src/calculatorImplementations.ts
import { Calculator, CalculatorError } from './calculator';

// 基础计算器实现
export class BasicCalculator implements Calculator {
    addFn(a: number, b: number): number {
        return a + b;
    }

    subFn(a: number, b: number): number {
        return a - b;
    }

    mulFn(a: number, b: number): number {
        return a * b;
    }

    divFn(a: number, b: number): number {
        if (b === 0) {
            throw new CalculatorError('除数不能为零');
        }
        return a / b;
    }
}

// 高级计算器实现
export class AdvancedCalculator implements Calculator {
    addFn(a: number, b: number): number {
        console.log(`[ADVANCED] 加法: ${a.toFixed(2)} + ${b.toFixed(2)}`);
        return a + b;
    }

    subFn(a: number, b: number): number {
        console.log(`[ADVANCED] 减法: ${a.toFixed(2)} - ${b.toFixed(2)}`);
        return a - b;
    }

    mulFn(a: number, b: number): number {
        console.log(`[ADVANCED] 乘法: ${a.toFixed(2)} * ${b.toFixed(2)}`);
        return a * b;
    }

    divFn(a: number, b: number): number {
        if (b === 0) {
            throw new CalculatorError('除数不能为零');
        }
        console.log(`[ADVANCED] 除法: ${a.toFixed(2)} / ${b.toFixed(2)}`);
        return a / b;
    }

    // 额外功能
    powFn(base: number, exponent: number): number {
        return Math.pow(base, exponent);
    }
}
